### PR TITLE
Support `@httpResponseCode` over newtypes

### PR DESCRIPTION
**Pull request by miguel-vila**
not sure if there was a reason to miss this case, let me know what u think

### CODE REVIEW

File: modules/core/src/smithy4s/http/internals/HttpResponseCodeSchemaVisitor.scala [hunk 1]
```diff
@@ -101,8 +101,18 @@ class HttpResponseCodeSchemaVisitor()
           case OptionalResponseCode(f) => f(value)
         }
     }
+  }
 
+  override def biject[A, B](
```
Comment (kubukoz) [inline modules/core/src/smithy4s/http/internals/HttpResponseCodeSchemaVisitor.scala:107]: **suggestion:** should we handle `def refine` as well?

Comment (Baccata) [reply modules/core/src/smithy4s/http/internals/HttpResponseCodeSchemaVisitor.scala:107]: Might as well ^^

**Comment by Baccata** [comment]
> not sure if there was a reason to miss this case, let me know what u think

No particular reason, sometimes we just miss things

File: modules/core/src/smithy4s/http/internals/HttpResponseCodeSchemaVisitor.scala (hunk 1 above)
Comment (miguel-vila) [reply modules/core/src/smithy4s/http/internals/HttpResponseCodeSchemaVisitor.scala:107]: done at [2dfd250](https://github.com/disneystreaming/smithy4s/pull/1377/commits/2dfd25048c65b2b4a46a90e7b692a55b69c44f4f)

**Review by kubukoz — APPROVED** [review]

**Comment by kubukoz** [comment]
gah, I forgot the CHANGELOG. @miguel-vila please check the PR template next time 😅

**Pull request merged**